napari/layers/utils/array_interface.py

```diff
@@ -0,0 +1,21 @@
+import sys
+from typing import Tuple, Union
+
+import numpy as np
+
+if sys.version_info.minor < 8:
+    from typing_extensions import Protocol
+else:
+    from typing import Protocol
+
+
+class ArrayInterface(Protocol):
+    shape: Tuple[int] = (0,)
+    dtype: type = np.uint8
+    ndim: int = 1
```

Comment:
```suggestion
    @property
    def ndim(self) -> int: ...
```
otherwise suggests that `ndim` is writable and will lead to this mypy error if someone implements as a property:
<img width="485" alt="Screen Shot 2020-04-30 at 12 05 35 PM" src="https://user-images.githubusercontent.com/1609449/80732905-f8ce6180-8ada-11ea-9abb-ebbc0264c42e.png">

Reply:
changed